Theme docs + minor polishments (#5360)
vuestorefront/vue-storefront: packages/core/docs/README.md

 - __PWA__ : _PWA_ is the thing of future, it is designed to give best performance even on laggy mobile devices with native-like features from the beginning in order to satisfy your customers who are curious, rich, and impatient.
 - __framework__ : Even though _Vue Storefront_ is a set of atomic features, they should be guided as one by the conductor. We do so by gluing them through _interfaces_ and _factories_. We will dig this further later, but one thing to take away, they are quite flexible glues.

## Problems Vue Storefront will solve for you

The main purpose of any software is to solve problems. Vue Storefront is no different. We're doing our best to find common issues in eCommerce space (and in our software!) and find viable solutions to them. Below you can find just a few of the issues Vue Storefront will solve for you.

### Long time to market

> Headless eCommerce frontends are complex and devloping them can take a lot of time. 

Reviewer: ```suggestion
Headless eCommerce frontends are complex and developing them can take a lot of time. 
```

> **Solution** With Vue Storefront you're getting a working headless frontend connected to eCommerce, CMS and other third-party services of your choice out of the box along with a hundreds of ready-to-use Vue Storefront and Nuxt modules for all common functionalities! Because of that you're saving hundreds (or even thousands) of working hours so you can focus more on details and polishments while leaving the heavy lifting to us!

Reviewer 2: devloping > developing

> **Solution** With Vue Storefront you're getting a working headless frontend connected to eCommerce, CMS and other third-party services of your choice out of the box along with a hundreds of ready-to-use Vue Storefront and Nuxt modules for all common functionalities! Because of that you're saving hundreds (or even thousands) of working hours so you can focus more on details and polishments while leaving the heavy lifting to us!

Reviewer 2: polishments > polishing

> **Solution** With Vue Storefront you're getting a working headless frontend connected to eCommerce, CMS and other third-party services of your choice out of the box along with a hundreds of ready-to-use Vue Storefront and Nuxt modules for all common functionalities! Because of that you're saving hundreds (or even thousands) of working hours so you can focus more on details and polishments while leaving the heavy lifting to us!

Author: @Fifciu I refer to nuxt modules as well

> **Solution** With Vue Storefront you're getting a working headless frontend connected to eCommerce, CMS and other third-party services of your choice out of the box along with a hundreds of ready-to-use Vue Storefront and Nuxt modules for all common functionalities! Because of that you're saving hundreds (or even thousands) of working hours so you can focus more on details and polishments while leaving the heavy lifting to us!

Author: Btw @ThermScissorpunch and @Fifciu please use suggestions feature for such stuff - it is much easier to apply them.

Also thanks for comments ;)

- Pages that were already visited are cached so next time user enters them they will load instantly.
- Resources that might be needed in the future are precached so when the user clicks the link it loads immediately.
- When network connectivity is slow or temporarily lost you can stil browse the product catalog. In many cases its unnoticeable for the user.

> - Majority of JavaScript code is hosted and executed on the server side so your users are downloading only the code that drives the UI. Because of this the page renders much faster and is much lighter compared to traditional SPA.

Author: ```suggestion
- Majority of JavaScript code is hosted and executed on the server side so your users download only the code that drives the UI. Because of this the page renders much faster and is much lighter compared to traditional SPA.
```

How frustrated was it when you learned the other backend platform you didn't choose turned out better solution for your business, a lot better on many levels, but the cost of switching is even greater than the benefit of it you just learned? You were literally locked-in by tentative choices you made while you were naive.

> **Solution** Vue storefront is backend-agnostic which means all eCommerce backends are integrated on the frontend under common interfaces. All these technologies are completely different on the backend but they're very similar from the frotnend perspective. We just made use of that fact and came up with abstractions that will make your migrations painless and almost instantaneous.

Reviewer: ```suggestion
**Solution** Vue Storefront is backend-agnostic which means all eCommerce backends are integrated on the frontend under common interfaces. All these technologies are completely different on the backend but they're very similar from the frontend perspective. We just made use of that fact and came up with abstractions that will make your migrations painless and almost instantaneous.
```

- Ability to work with any eCommerce platform, CMS, ERP, PIM **under common, agnostic API**
- Server Side Rendereing
- Progressive Web App features

> - Limitless flexibility in changing third-party services thanks to agnostic data formats

Reviewer: Unlimited?

> - Limitless flexibility in changing third-party services thanks to agnostic data formats

Author: ```suggestion
- Unlimited flexibility in changing third-party services thanks to agnostic data formats
```
